### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by drahnr** [rpm-rs/rpm#31] [General]
The roadmap forward, especially features and usability improvements.

**Comment by TommyLike** [comment]
Support DSA signature

**Reply by drahnr** [reply]
I am not entirely sure how we want DSA. The original proclaimed goal of rpm-rs was to deliberately not support each and every feature of the crufty c rpm impl. DSA is likely to be canned in the next FIPS update. I am hence not sure how relevant it is for us to implement. If there is a showcase of this being used in practice within the last 5 years, it seems reasonable to support, if it's only older versions, I find it debatable to spend impl and ongoing maintenence effort.

**Reply by dralley** [reply]
@TommyLike When you say DSA, do you mean legacy DSA or EdDSA / Ed25519?  RPM re-used the DSA tag within the header to store EdDSA / Ed25519 signatures.

DSA as @drahnr mentioned is very much obsolete.  EdDSA is not and contributions would be welcome there.

**Reply by dralley** [reply]
https://github.com/rpm-rs/rpm/pull/146

**Comment by cmeister2** [comment]


**Reply by cmeister2** [reply]
WIP: https://github.com/rpm-rs/rpm/pull/40

**Comment by cmeister2** [comment]
Don't require the whole file to be parsed if you just want the metadata. #6

**Reply by TommyLike** [reply]
Is there any way to find out the precise size of the signature and header section? I mean the lead section is fixed size but how could tell the size of headers?

**Reply by drahnr** [reply]
I think should go into the issues #6 itself rather than in the roadmap discussion :)

**Reply by dralley** [reply]
@TommyLike This is possible now fyi
